### PR TITLE
table: avoid write untouched index in auto commit and avoid get untouched index key from tikv

### PR DESCRIPTION
**Pull request by crazycs520**
### What problem does this PR solve? 
PR #11702 optimized union-scan performance, but has performance  regression when update with untouch index. Below is the cpu.profile:

image

As you can see, when update untouched index, we no need to fetch index data from TiKV again.
And in the auto-commit transaction, we no need to write  the untouched index to txn membuffer, because txn will commit after update statement, we won't read from txn menbuffer anymore.

### Check List 

Tests 

 - Unit test.

Code changes

 - Has exported function/method change

### CODE REVIEW

**Comment by crazycs520** [comment]
@tiancaiamao @coocood @jackysp PTAL

File: table/tables/tables.go [hunk 1]
```diff
@@ -388,6 +388,10 @@ func (t *tableCommon) rebuildIndices(ctx sessionctx.Context, rm kv.RetrieverMuta
 			untouched = false
 			break
 		}
+		// If txn is auto commit and index is untouched, no need to write index value.
+		if untouched && !ctx.GetSessionVars().InTxn() {
+			return nil
```
Comment (jackysp) [inline table/tables/tables.go]: why return nil here? the following indices may need to be created.

Comment (crazycs520) [reply table/tables/tables.go]: The untouched index value must already exist in tikv, no need to create again.

**Comment by tiancaiamao** [comment]
LGTM

**Comment by coocood** [comment]
LGTM

**Pull request merged**